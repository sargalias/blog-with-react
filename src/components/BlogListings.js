import React from 'react';
import BlogListing from './BlogListing';

const BlogListings = ({ blogPosts = []}) => (
  blogPosts.length > 0 && (
    <div className="blog-listings mt-4">
      { blogPosts.map((blogPost) => (
        <BlogListing blogPost={blogPost} key={blogPost.postId} />
      ))}
    </div>
  )
);

export default BlogListings;
