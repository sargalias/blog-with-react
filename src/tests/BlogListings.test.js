import React from 'react';
import { shallow } from 'enzyme';
import BlogListings from '../components/BlogListings';
import blogPosts from './fixtures/blogPosts';

describe('BlogListings', () => {
  test('Should render correctly without props', () => {
    const wrapper = shallow(<BlogListings />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Should render correctly with props', () => {
    const wrapper = shallow(<BlogListings blogPosts={blogPosts} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Should be empty with no props', () => {
    const wrapper = shallow(<BlogListings />);
    expect(wrapper.html()).toBeNull();
  });

  test('Should be empty with prop blogPosts=[]', () => {
    const wrapper = shallow(<BlogListings blogPosts={[]} />);
    expect(wrapper.html()).toBeNull();
  });

  test('Should be non-empty with props', () => {
    const wrapper = shallow(<BlogListings blogPosts={blogPosts} />);
    expect(wrapper.html()).toBeTruthy();
  });

  test('Should have correct number of BlogListing children', () => {
    const length = 3;
    const wrapper = shallow(<BlogListings blogPosts={blogPosts.slice(0, length)} />);
    expect(wrapper.find('BlogListing').length).toBe(3);
  });

  test('Should call BlogListing children with correct props', () => {
    const legnth = 3;
    const wrapper = shallow(<BlogListings blogPosts={blogPosts.slice(0, length)} />);
    const blogListingList = wrapper.find('BlogListing');
    blogListingList.forEach((blogListing, i) => {
      expect(blogListing.props().blogPost).toEqual(blogPosts[i]);
    });
  });
});
