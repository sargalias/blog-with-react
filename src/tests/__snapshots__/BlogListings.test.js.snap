// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogListings Should render correctly with props 1`] = `
<div
  className="blog-listings mt-4"
>
  <BlogListing
    blogPost={
      Object {
        "category": "Food",
        "date": "1970-01-01T00:00:00.000Z",
        "excerpt": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Est fugiat harum obcaecati pariatur sint tempore unde. Aperiam commodi distinctio earum eligendi, enim itaque laudantium magni neque porro temporibus vitae voluptatum?",
        "imageURL": "https://via.placeholder.com/350x350",
        "postId": "1",
        "title": "Test title",
        "user": Object {
          "displayName": "User displayName 1",
          "id": 1,
        },
      }
    }
    key="1"
  />
  <BlogListing
    blogPost={
      Object {
        "category": "Travel",
        "date": "1970-01-06T00:00:00.000Z",
        "excerpt": "Test excerpt 2",
        "imageURL": "https://via.placeholder.com/200x200",
        "postId": "2",
        "title": "Test title 2",
        "user": Object {
          "displayName": "User displayName 2",
          "id": 2,
        },
      }
    }
    key="2"
  />
  <BlogListing
    blogPost={
      Object {
        "category": "Food",
        "date": "1969-12-27T00:00:00.000Z",
        "excerpt": "3 Lorem ipsum dolor",
        "imageURL": "https://via.placeholder.com/250x250",
        "postId": "3",
        "title": "Test title 3",
        "user": Object {
          "displayName": "User displayName 3",
          "id": 3,
        },
      }
    }
    key="3"
  />
</div>
`;

exports[`BlogListings Should render correctly without props 1`] = `""`;
